### DISCUSSION TITLE
deployment not working

### DISCUSSION THREAD

**Discussion by jamiepratt** [JamesIves/github-pages-deploy-action#1190] [Q&A]
Hi,

I'm trying to do an automatic custom build of some static pages to serve through Github pages.

[The workflow](https://github.com/jamiepratt/jointprob-clerk/blob/master/.github/workflows/main.yml) runs without errors and the deployment seems to work:

<img width="813" alt="Screenshot 2022-08-12 at 10 41 41" src="https://user-images.githubusercontent.com/558780/184321473-cdad1956-2348-4c81-8023-118fb7d3d111.png">

And I see a gh-pages branch has been created and [an index.html](https://github.com/jamiepratt/jointprob-clerk/blob/gh-pages/index.html) has been added to it.

But my pages for some reason were not served by github.io: 

https://jamiepratt.github.io/jointprob-clerk/

Showed a 404.

The pages were there in the gh-pages branch.

So I switched to the 'deploy from branch' option in "Settings -> Pages" of my repository.

The files did get deployed. But they are not updated when I do a new push of my master branch.

I have switched back to source "github action" to serve the files. And the files are getting deployed to the gh-pages branch but I can only update what I see at https://jamiepratt.github.io/jointprob-clerk/ by going to trigger the workflow action "pages-build-deployment" which I can re-run clicking through from here:

<img width="1139" alt="Screenshot 2022-08-12 at 16 43 24" src="https://user-images.githubusercontent.com/558780/184380080-0f83d572-1303-4c26-a22e-dcca0e38e5c3.png">

Jamie

**Comment by jamiepratt** [comment]
Continuing to have problems with this. Using this action I don't see the content served by github.io updating although I do see the files committed to gh-pages branch of my repo.

**Comment by jamiepratt** [comment]
This is what I did in the end:

https://github.com/jamiepratt/jointprob-clerk/blob/master/.github/workflows/main.yml

I based my workflow on another:

https://github.com/actions/starter-workflows/blob/main/pages/jekyll-gh-pages.yml

And just replaced the jekyll build with the clojure and clerk build process.

Your github-pages-deploy-action didn't work for me.

**Comment by inorganik** [comment]
I had this same issue, and I think it's because of a misunderstanding of how this action works. The action gets configured in your workflow. But you _do not_ set "Github actions" as a source for Github pages.

Make sure the pages source is "Deploy from branch" with the "gh-pages" branch selected. Everytime you push, (or whatever trigger you configured), this action overwrites the contents of the gh-pages branch, thereby deploying the site. 

This action works differently than most deploy actions because it simply sets the contents of the gh-pages branch, instead of generating and uploading an artifact.

**Reply by ppvan** [reply]
I think there should be more clear in the example. I just try it out, 3 hours to find this page
Deploy from a branch did fix the issue, thank you

**Reply by francisbarton** [reply]
OMG I cannot believe it! Thank you @inorganik . I have been going crazy for weeks trying to work out why my successful Actions runs resulted in no GH Pages site actually being produced. It's really not clear in the docs that you should keep the setting as "Deploy from branch" - it seems obvious that using this action means you should use the "Github actions" option.

**Reply by greggman** [reply]
>  it seems obvious that using this action means you should use the "Github actions" option.

Doesn't seem obvious at all, at least to me. The action is called "deploy" not "build-to-folder-of-your-choice" and then having to actually deploy using some other action.

This is not a dis on this wonderful, useful script. But, it would be helpful if the docs were made clearer

**Reply by JamesIves** [reply]
I've gone ahead and clarified this in the README under the getting started section. The actual GitHub Pages product has evolved since this action was originally created 5 years ago, I had no idea this was even an option you could choose between.

If there's something that is not clear in the README I'm more than happy to make adjustments based on suggestions!
